octoprint_psucontrol/__init__.py

```diff
@@ -118,7 +126,10 @@ def on_settings_initialized(self):
 
         self._checkPSUTimer = RepeatedTimer(5.0, self.check_psu_state, None, None, True)
         self._checkPSUTimer.start()
-
+	
+	self._checkButtonTimer = RepeatedTimer(3.0, self.check_button_state, None, None, True)
+        self._checkButtonTimer.start()

```

Comment:
You should use a GPIO interrupt instead of polling the pin state every couple of seconds. gives you instant reaction, other than pressing the button between 0 and 3 seconds.